pkg/expr/ml.go

```diff
@@ -92,24 +78,39 @@ func (m *MLNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s *
 	// Execute the command and provide callback function for sending a request via plugin API.
 	// This lets us make commands abstracted from peculiarities of the transfer protocol.
 	data, err := m.command.Execute(timeRange.From, timeRange.To, func(method string, path string, payload []byte) (response.Response, error) {
-		crReq := &backend.CallResourceRequest{
-			PluginContext: pCtx,
-			Path:          path,
-			Method:        method,
-			URL:           path,
-			Headers:       make(map[string][]string, len(m.request.Headers)),
-			Body:          payload,
+		validate := func(ctx context.Context, pCtx backend.PluginContext) error {
+			// Plugin must be initialized by the admin first. That will create service account, and update plugin settings so all requests can use it.
+			// Fail if it is not initialized.
+			if pCtx.AppInstanceSettings == nil || !jsoniter.Get(pCtx.AppInstanceSettings.JSONData, "initialized").ToBool() {
+				return errMLPluginDoesNotExist
+			}
+
+			return nil
+		}
```

Comment:
I think that's fine if you put it here. The idea of the check above was to avoid unnecessary marshaling.